yt_idefix/data_structures.py

```diff
@@ -506,7 +510,15 @@ def _parse_header_file(self):
                 if unit_match is not None:
                     unit = unit_match.group(1).lower() + "_unit"
                     expr = unit_match.group(2)
-                    expr = re.sub(constexpr, self._get_constants, expr)
+                    # Before evaluate the expression, replace the input parameters,
+                    # pre-defined constants, code units and sqrt function
+                    # that cannot be resolved.
+                    expr = re.sub(
+                        r"g_inputParam\[(\w+)\]", self._get_input_parameter, expr
+                    )
+                    expr = re.sub(r"CONST_\w+", self._get_constants, expr)
+                    expr = re.sub(r"UNIT_(\w+)", self._get_unit, expr)
+                    expr = re.sub(r"sqrt", "np.sqrt", expr)
```

Comment:
Does order matter ? It doesn't seem to, but in any case please add a comment to make it explicit.

Reply:
Okay, I'll improve it.